---
title: "Circular Genome View"
author: "Gabriel Falcao Alencar"
date: "February, 2018"
output: 
  html_document: 
    keep_md: yes
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

Load the necessary packages
```{r packages}
library(circlize)
```

Next step, we will generate a plain graph to see what we need to alter.

```{r first}
circos.initializeWithIdeogram()
text(0, 0, "default", cex = 1)
circos.info()
circos.clear()
```

Things that I want to change:  
* Space between the chromosomes;  
* Start on chromosome 1;  
* Add points to correct places;  

Start altering the location and space

```{r local-space}
circos.par("start.degree" = 90)
circos.par("gap.degree" = rep(c(2, 4), 12))
circos.initializeWithIdeogram()
circos.clear()
```

Testing with random data before I make my final figure

```{r randon-data-graph}
circos.par("start.degree" = 90)
circos.par("gap.degree" = rep(c(2, 4), 12))
circos.initializeWithIdeogram()

bed = generateRandomBed(nr = 20, nc = 0)
circos.genomicPosTransformLines(bed, posTransform = posTransform.default, horizontalLine = "top", track.height = 0.1)

circos.genomicTrackPlotRegion(bed, ylim = c(0, 1), panel.fun = function(region, value, ...) {
    circos.genomicText(region, value, y = 1, adj = c(0, 0.5), labels = "gene", facing = "reverse.clockwise", niceFacing = TRUE,
        posTransform = posTransform.default)
}, bg.border = NA)

circos.clear()
```

Next things that I want to alter:  
1. Can I remove the "size" of the chromosome from the plot?  
2. Should I make each chromosome a solid color?

Add data that we need:

```{r load-data}
snps.up.2010 <- read.csv("./create-circular-genomes_files/SNP-data/up-to-2010.csv", header=TRUE)
snps.2011.to.15 <- read.csv("./create-circular-genomes_files/SNP-data/2011-2015.csv", header=TRUE)
snps.2016 <- read.csv("./create-circular-genomes_files/SNP-data/2016.csv", header=TRUE)
snps.2017 <- read.csv("./create-circular-genomes_files/SNP-data/2017.csv", header=TRUE)

snps.up.2010 <- unique(snps.up.2010[order(snps.up.2010$chr),])
snps.2011.to.15 <- unique(snps.2011.to.15[order(snps.2011.to.15$chr),])
snps.2016 <- unique(snps.2016[order(snps.2016$chr),])
snps.2017 <- unique(snps.2017[order(snps.2017$chr),])
```

Now lets plot up to 2010
```{r up-2010}
circos.par("start.degree" = 90)
circos.par("gap.degree" = rep(c(2, 4), 12))
circos.initializeWithIdeogram()

circos.genomicPosTransformLines(snps.up.2010, posTransform = posTransform.default, horizontalLine = "top", track.height = 0.1)

circos.genomicTrackPlotRegion(snps.up.2010, ylim = c(0, 1), panel.fun = function(region, value, ...) {
    circos.genomicText(region, value, y = 1, adj = c(0, 0.5), labels = "loci", facing = "reverse.clockwise", niceFacing = TRUE,
        posTransform = posTransform.default)
}, bg.border = NA)

circos.clear()
```

Plotting up to 2015
```{r up-2015}
circos.par("start.degree" = 90)
circos.par("gap.degree" = rep(c(2, 4), 12))
circos.initializeWithIdeogram()

up.2015 <- merge(snps.up.2010, snps.2011.to.15, all=TRUE)
circos.genomicPosTransformLines(up.2015, posTransform = posTransform.default, horizontalLine = "top", track.height = 0.1)

circos.genomicTrackPlotRegion(up.2015, ylim = c(0, 1), panel.fun = function(region, value, ...) {
    circos.genomicText(region, value, y = 1, adj = c(0, 0.5), labels = "loci", facing = "reverse.clockwise", niceFacing = TRUE,
        posTransform = posTransform.default)
}, bg.border = NA)

circos.clear()
```

Now lets plot up to 2016
```{r up-2016}
circos.par("start.degree" = 90)
circos.par("gap.degree" = rep(c(2, 4), 12))
circos.initializeWithIdeogram()

up.2016 <- merge(up.2015, snps.2016, all=TRUE)
circos.genomicPosTransformLines(up.2016, posTransform = posTransform.default, horizontalLine = "top", track.height = 0.1)

circos.genomicTrackPlotRegion(up.2016, ylim = c(0, 1), panel.fun = function(region, value, ...) {
    circos.genomicText(region, value, y = 1, adj = c(0, 0.5), labels = "loci", facing = "reverse.clockwise", niceFacing = TRUE,
        posTransform = posTransform.default)
}, bg.border = NA)

circos.clear()
```

Now lets plot up to 2017
```{r up-2017}
circos.par("start.degree" = 90)
circos.par("gap.degree" = rep(c(2, 4), 12))
circos.initializeWithIdeogram()

up.2017 <- merge(up.2016, snps.2017, all=TRUE)
circos.genomicPosTransformLines(up.2017, posTransform = posTransform.default, horizontalLine = "top", track.height = 0.1)

circos.genomicTrackPlotRegion(up.2017, ylim = c(0, 1), panel.fun = function(region, value, ...) {
    circos.genomicText(region, value, y = 1, adj = c(0, 0.5), labels = "loci", facing = "reverse.clockwise", niceFacing = TRUE,
        posTransform = posTransform.default)
}, bg.border = NA)

circos.clear()
```